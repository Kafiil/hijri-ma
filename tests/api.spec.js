const { fetchData } = require('../src/api');
const config = require('../src/config');

describe.only('Steps to get the text', () => {
  it('Should return a valid result', async () => {
    const result = await fetchData(config);
    expect(result).toContain('هـ');
  });

  it('Should return a response similar to snapshot', async () => {
    const result = await fetchData(config);
    expect(result).toMatchSnapshot();
  });

  it('Should throw an error', async () => {
    try {
      const result = await fetchData({
        websiteUrl: 'http://localhost:33/',
        textNode: 'not a valid node',
      });
    } catch (error) {
      expect(error).toBe('Some Error');
    }
  });
});
