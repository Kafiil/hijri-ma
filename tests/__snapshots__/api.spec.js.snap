// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps to get the text Should return a response similar to snapshot 1`] = `"السبت 28 شوّال 1441هـ الموافق لـ 20 يونيو 2020"`;
